### PR TITLE
UIEH-1052: Jest+RTL > Test PeerReviewedField, PublicationTypeField, PublisherNameField, TitleListItem and Toast

### PR DESCRIPTION
**Pull request by mariiaschv**
## Purpose

- Refactoring
- Add tests for PeerReviewedField, PublicationTypeField, TitleListItem, Toast
- Tests for PublisherNameField have been already written

Issue: [UIEH-1052](https://issues.folio.org/browse/UIEH-1052)

### CODE REVIEW

File: src/components/title-list-item/title-list-item.test.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+};
+
+const testBasicProps = {
+  ...testPropsWithoutItem,
+  item: {
+    contributors: [
+      {
+        type: 'test-contributor-type',
+        contributor: 'test-contributor-name',
+      }
```
Comment (Vladyslav-Velytskyi) [inline src/components/title-list-item/title-list-item.test.js:32]:
```suggestion
      },
```

File: src/components/title-list-item/title-list-item.test.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    ],
+    identifiers: [
+      {
+        id: 'test-id',
+        subtype: 'test-subtype',
+        type: 'test-type',
+      }
```
Comment (Vladyslav-Velytskyi) [inline src/components/title-list-item/title-list-item.test.js:39]:
```suggestion
      },
```

File: src/components/title-list-item/title-list-item.test.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    });
+
+    it('should display hidden label', () => {
+      const testExtendedProps = {
+        ...testBasicProps,
+        item: {
+          ...testBasicProps.item,
+          visibilityData: {
+            isHidden: true,
+          }
```
Comment (Vladyslav-Velytskyi) [inline src/components/title-list-item/title-list-item.test.js:171]:
```suggestion
          },
```

File: src/components/title-list-item/title-list-item.test.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+      ...testBasicProps,
+      showIdentifiers: true,
+    };
+
+    const { getByText } = renderTitleListItem(testExtendedProps);
+
+    expect(getByText('IdentifiersList component')).toBeDefined();
+  });
+
+  describe('item metadata class', () => {
```
Comment (Vladyslav-Velytskyi) [inline src/components/title-list-item/title-list-item.test.js:140]: I'm not sure that you need this `describe`. Actually, describe should starts with word `when` and logically separate tests. I've recomended to add all props that you add to `testExtendedProps` in test set as default. In this case you don't need to set this data each time when you write test.

File: src/components/title/_fields/publication-type/publication-type-field.js [hunk 5]
```diff
@@ -1,15 +1,24 @@
...
     <div data-test-eholdings-publication-type-field>
       <Field
         name="publicationType"
         component={Select}
-        label={<FormattedMessage id="ui-eholdings.title.publicationType" />}
+        label={label}
+        aria-label={label}
+        data-testid="publication-type-field"
       >
         <FormattedMessage id="ui-eholdings.filter.pubType.audioBook">
```
Comment (Vladyslav-Velytskyi) [inline src/components/title/_fields/publication-type/publication-type-field.js:23]: Could you also fix this duplication like it done for `PackageContentTypeField`?

File: src/components/title/_fields/publication-type/publication-type-field.test.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    />
+  );
+
+  it('should render peer reviewed field', () => {
+    const { getByText } = renderPublicationTypeField();
+
+    expect(getByText('ui-eholdings.title.publicationType')).toBeDefined();
+  });
+
+  it('should display Audiobook option', () => {
```
Comment (Vladyslav-Velytskyi) [inline src/components/title/_fields/publication-type/publication-type-field.test.js:20]: Just suggestion. You could write this tests in cycle.
For example
```
Object.values(objOfTypes).forEach(value => {
  it('should display option', () => {
     const { getByText } = renderPublicationTypeField();
   
     expect(getByText(`ui-eholdings.filter.pubType.${value}`)).toBeDefined();  
  });
});
```
or
```
it('should display options', () => {
  const { getByText } = renderPublicationTypeField();
   
  Object.values(objOfTypes).forEach(value => {
    expect(getByText(`ui-eholdings.filter.pubType.${value}`)).toBeDefined(); 
  });
});
```

**Review by Vladyslav-Velytskyi — CHANGES_REQUESTED** [review]

**Review by Vladyslav-Velytskyi — APPROVED** [review]
LGTM!

File: src/components/title-list-item/title-list-item.test.js [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+      isHidden: true,
+    },
+  },
+};
+
+describe('Given TitleListItem', () => {
+  const mockOnClick = jest.fn();
+
+  const renderTitleListItem = ({ ...props }) => render(
+    <Router history={routerHistory}>
```
Comment (BogdanDenis) [inline src/components/title-list-item/title-list-item.test.js:76]: You can use `<Harness>` component from `test/jest/helpers/harness`. It already has a router and other mocked components

**Review by BogdanDenis — CHANGES_REQUESTED** [review]

**Review by BogdanDenis — APPROVED** [review]

**Pull request merged**